### PR TITLE
SOX-21 table pull

### PR DESCRIPTION
**Pull request by zajca**
Jira: SOX-21
KBC: https://github.com/keboola/connection/pull/4464

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Review by romanbracinik — COMMENTED** [review]
Mozno by bolo dobre otestovat aj ine akcie ako len update metadat, napr pridanie alebo odobranie stlpcov. Mozno dalsi test na typovanu tabulku a tak. Inac testy presli

**Comment by zajca** [comment]
> Mozno by bolo dobre otestovat aj ine akcie ako len update metadat, napr pridanie alebo odobranie stlpcov. Mozno dalsi test na typovanu tabulku a tak. Inac testy presli

Přidal sem asserty na eventy. přidal sem přidání odebrání sloupce a test na tabulku s typama.

**Review by romanbracinik — CHANGES_REQUESTED** [review]
Jedna mala drobnost k eventam a bude to vsetko, PR connection je cajk

File: tests/Backend/SOX/PullTableTest.php [hunk 1]
```diff
@@ -84,10 +86,37 @@ public function testPullTableFromDefaultBranch(): void
             'token' => STORAGE_API_DEVELOPER_TOKEN,
             'url' => STORAGE_API_URL,
         ]);
+        // set tokenId to fetch events
+        $this->tokenId = explode('-', STORAGE_API_DEVELOPER_TOKEN)[1];
 
+        $runId = $privilegedClient->generateRunId();
+        $branchClient->setRunId($runId);
+
+        $event = $this->createAndWaitForEvent(
```
[the remark this answers is not in the document]
Comment (zajca) [reply tests/Backend/SOX/PullTableTest.php:94]: Na to si snad nezvyknu :D

**Review by romanbracinik — APPROVED** [review]
cool 👍

**Pull request merged**